packages/typespec-client-generator-core/src/types.ts

```diff
@@ -195,12 +189,20 @@ function getScalarKind(scalar: Scalar): SdkBuiltInKinds {
     case "decimal":
     case "plainDate":
     case "plainTime":
+    case "arm-id":
```

Comment:
Sorry a lot of those are not built-in the compiler, they are in azure-core so you have to check the right namespace